lib/virtio/virtqueue.c

```diff
@@ -88,6 +88,24 @@ int virtqueue_create(struct virtio_device *virt_dev, unsigned short id,
 	return (status);
 }
 
+struct virtqueue *
+virtqueue_allocate(struct virtio_device *vdev, unsigned int num_descs)
+{
+	struct virtqueue *vq;
+	uint32_t vq_size = sizeof(*vq);
+
+	/* vq_desc_extra is not used in virito backend */
```

Comment:
I think this is partially an optimization related to support for VIRTIO_DEV_GUEST v HOST.  Since there are various APIs one should not call if VIRTIO_HOST_GUEST is set, and thus we need to probably both check in those APIs and support building them out as a compile option.

Reply:
i send a patch that reworks this in my daily poll request

Reply:
Arnaud's patches have been folded into your tree already.